test/testutils/util.go

```diff
@@ -696,3 +738,194 @@ func SetLwsPartition(ctx context.Context, k8sClient client.Client, lws *leaderwo
+	gomega.Expect(ownerRefMap["apiVersion"]).To(gomega.Equal("v1"))
+	gomega.Expect(ownerRefMap["kind"]).To(gomega.Equal("Pod"))
+	gomega.Expect(ownerRefMap["name"]).To(gomega.Equal(leaderPod.Name))
+	gomega.Expect(ownerRefMap["uid"]).To(gomega.Equal(string(leaderPod.UID)))
+
+	controller, found, err := unstructured.NestedBool(ownerRefMap, "controller")
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	gomega.Expect(found).To(gomega.BeTrue())
+	gomega.Expect(controller).To(gomega.BeTrue())
+}
+
+// validatePodGroupSpec validates the PodGroup spec using unstructured access
+func validatePodGroupSpec(podGroup *unstructured.Unstructured, lws *leaderworkerset.LeaderWorkerSet) {
+	minMember, found, err := unstructured.NestedInt64(podGroup.Object, "spec", "minMember")
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	gomega.Expect(found).To(gomega.BeTrue())
+
+	// For LeaderReady startup policy, minMember should be 1 (leader only)
+	// For LeaderCreated startup policy, minMember should be size (leader + workers)
+	expectedMinMember := int64(*lws.Spec.LeaderWorkerTemplate.Size)
+	if lws.Spec.StartupPolicy == leaderworkerset.LeaderReadyStartupPolicy {
+		expectedMinMember = 1
+	}
+
+	gomega.Expect(minMember).To(gomega.Equal(expectedMinMember))
+
+	// Validate minResources - should be the same for both LeaderCreated and LeaderReady
+	// as it represents total resources needed for the entire PodGroup [1 Leader + (size-1) Worker pods]
+	expectedMinResources := utils.CalculatePGMinResources(lws)
+
+	minResourcesMap, found, err := unstructured.NestedMap(podGroup.Object, "spec", "minResources")
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	gomega.Expect(found).To(gomega.BeTrue())
+
+	// Validate each resource type in minResources
+	for resourceName, expectedQuantity := range expectedMinResources {
+		actualQuantityStr, exists := minResourcesMap[string(resourceName)]
+		gomega.Expect(exists).To(gomega.BeTrue(), "Resource %s should exist in minResources", resourceName)
+
+		actualQuantity, err := resource.ParseQuantity(actualQuantityStr.(string))
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(actualQuantity.Equal(expectedQuantity)).To(gomega.BeTrue(),
+			"Resource %s: expected %s, got %s", resourceName, expectedQuantity.String(), actualQuantity.String())
+	}
+}
+
+// ExpectValidPodGroups validates that PodGroups are correctly created for a LeaderWorkerSet with proper owner references and spec configuration.
+// When expectedCount is 0, it validates that no PodGroups exist for the LWS, checking cleanup.
+// When expectedCount > 0, it validates both count and detailed configuration of each PodGroup.
+func ExpectValidPodGroups(ctx context.Context, k8sClient client.Client, provider schedulerprovider.ProviderType, lws *leaderworkerset.LeaderWorkerSet, expectedCount int) {
+	gvk := getPodGroupGVK(provider)
+	if gvk.Empty() {
+		ginkgo.Skip("Unsupported scheduler provider for PodGroup validation")
```

Comment:
I think we should return error here? Empty gvk is not allowed I think.

Reply:
OK I have changed to `ginkgo.Fail`, please check